bundles/http/Citrix/2ec16a1b-9423-45a6-8fd8-b0c42c916c0f/e40510b3-a760-43b4-ab16-56572dc6303b.js

```diff
@@ -0,0 +1,887 @@
+    }
+    // meetingset = new Set([meeting.id]);
+    await detectTypeChange(actionParameters,occurrences, dataStore);
+    await syncMeetings(dataStore, client, [{userid: actionParameters.Userid }]);
+    meetingset = new Set([actionParameters.meetingid]);
+    let meetingDetails = await syncMeetingsDetails(dataStore, client, meetingset);
+    await syncMeetingInvitations(dataStore, client,meetingDetails);
+}
+
+
+async function editRecurringMeeting({ dataStore, client, actionParameters }) {
+    let meetingset;
+    const request =await client.fetch(`meetings/${actionParameters.meetingid}?occurrence_id=${actionParameters.occurrence_id}`, {
+        method: "PATCH",
+        body: JSON.stringify({
+            start_time: actionParameters.start_time,
+            duration: actionParameters.duration
+        })
+    })
+    if (!(await request.ok)) {
+        throw new Error(JSON.stringify(await request.json()))
+    }
+    console.log()
+    syncMeetings(dataStore, client, [{userid: actionParameters.userid }]);
+    meetingset = new Set([actionParameters.meetingid]);
+    let meetingDetails = await syncMeetingsDetails(dataStore, client, meetingset);
+    await syncMeetingInvitations(dataStore, client,meetingDetails);
+
+}
+
+async function detectTypeChange(actionParameters, meetingDetail, dataStore) {
+    let change = false;
+    if (actionParameters.previous_type != actionParameters.type) {
+        if (actionParameters.type == 2) {
+            change = true
+        }
+        else if (actionParameters.type == 8) {
+            let onetime = {
+                "host_id": actionParameters.Userid,
+                "id": Number(actionParameters.meetingid),
+                "topic": actionParameters.topic,
+                "type": Number(actionParameters.type),
+                "visibility": false,
+            }
+            dataStore.save('meeting_details_one_time', onetime);
+        }
+    }
+    else {
+        if (actionParameters.pre_rtype != actionParameters.r_type) {
+            change=true
+        }
+    }
+
+    if(change){
+        let occurrenceData = meetingDetail.map(occurrence => {
+            return {
+                "duration": occurrence.duration,
+                "occurrence_id": Number(occurrence.occurrence_id),
+                "meetingId": Number(actionParameters.meetingid),
+                "start_time": new Date(occurrence.start_time),
+                "parent_id": Number(actionParameters.meetingid),
+                "visibility": false,
+            }
+        })
+        dataStore.save('meeting_details_recur_1', occurrenceData);
+        let recurring = {
+            "host_id": actionParameters.Userid,
+            "id": Number(actionParameters.meetingid),
+            "topic": actionParameters.topic,
+            "type": Number(actionParameters.type),
+            "visibility": false,
+        }
+        dataStore.save('meeting_details_recurring', recurring);

```

Comment:
```suggestion
        dataStore.save('meeting_details_recur_1',  meetingDetail.map(occurrence => ({
                duration: occurrence.duration,
                occurrence_id: occurrence.occurrence_id,
                meetingId: actionParameters.meetingid,
                start_time: new Date(occurrence.start_time),
                parent_id: actionParameters.meetingid,
                visibility: false,
            });
        dataStore.save('meeting_details_recurring', {
            host_id: actionParameters.Userid,
            id: actionParameters.meetingid,
            topic: actionParameters.topic,
            type: actionParameters.type,
            visibility: false,
        });
```
and this whole block could be in a separate function.
And this function can be called where `change=true` so handling `change` variable is not necessary.